pkgs/jnigen/lib/src/bindings/linker.dart

```diff
@@ -119,6 +120,29 @@ class _ClassLinker extends Visitor<ClassDecl, void> {
     final superclass = (node.superclass!.type as DeclaredType).classDecl;
     superclass.accept(this);
 
+    // Add all methods from the superinterfaces of this class.
+    final methodSignatures = <String>{};
+    for (final method in node.methods) {
+      methodSignatures.add(method.javaSig);
+    }
+    for (final interface in node.interfaces) {
+      interface.accept(typeLinker);
+      if (interface.type case final DeclaredType interfaceType) {
+        interfaceType.classDecl.accept(this);
+        for (final interfaceMethod in interfaceType.classDecl.methods) {
+          final clonedMethod =
+              interfaceMethod.clone(until: GenerationStage.linker);
```

Comment:
Why do you need to clone the method? Seems a bit expensive and error prone. In ffigen we just have multiple classes/interfaces hold refs to the same method object. If you really need the `method.classDecl` field etc, then maybe it would make sense to introduce a layer of indirection? Eg `ClassDecl` -> `MethodRef` -> `Method`. The `MethodRef` is where the class back-link would live, but all the important bits would stay in the `Method`, and then you'd only have to clone the `MethodRef`. That might also fix your CI failures, where you're getting multiple definitions of things.

Reply:
I need to clone the method because it could be a completely different method, for example `T foo()` of a `Foo<T>` when we do `implements Foo<String>` will become `String foo()`.